src/chainstate/stacks/auth.rs

```diff
@@ -21,44 +21,1431 @@ use net::StacksMessageCodec;
+                nonce: 456,
+                fields: vec![
+                    TransactionAuthField::Signature(TransactionPublicKeyEncoding::Compressed, MessageSignature::from_raw(&vec![0xff; 65])),
+                    TransactionAuthField::Signature(TransactionPublicKeyEncoding::Compressed, MessageSignature::from_raw(&vec![0xfe; 65])),
+                    TransactionAuthField::PublicKey(PubKey::from_hex("03ef2340518b5867b23598a9cf74611f8b98064f7d55cdb8c107c67b5efcbc5c77").unwrap())
+                ],
+                signatures_required: 2
+            }),
+            TransactionSpendingCondition::Singlesig(SinglesigSpendingCondition {
+                signer: Hash160([0x11; 20]),
+                hash_mode: SinglesigHashMode::P2WPKH,
+                key_encoding: TransactionPublicKeyEncoding::Compressed,
+                nonce: 345,
+                signature: MessageSignature::from_raw(&vec![0xfe; 65]),
+            }),
+            TransactionSpendingCondition::Multisig(MultisigSpendingCondition {
+                signer: Hash160([0x11; 20]),
+                hash_mode: MultisigHashMode::P2WSH,
+                nonce: 456,
+                fields: vec![
+                    TransactionAuthField::Signature(TransactionPublicKeyEncoding::Compressed, MessageSignature::from_raw(&vec![0xff; 65])),
+                    TransactionAuthField::Signature(TransactionPublicKeyEncoding::Compressed, MessageSignature::from_raw(&vec![0xfe; 65])),
+                    TransactionAuthField::PublicKey(PubKey::from_hex("03ef2340518b5867b23598a9cf74611f8b98064f7d55cdb8c107c67b5efcbc5c77").unwrap())
+                ],
+                signatures_required: 2
+            })
+        ];
+
+        for i in 0..spending_conditions.len() {
+            let spending_condition_bytes = spending_conditions[i].serialize();
+            let spending_condition_2_bytes = spending_conditions[(i+1) % spending_conditions.len()].serialize();
+
+            let auth_standard = TransactionAuth::Standard(spending_conditions[i].clone());
+            let mut auth_standard_bytes = vec![
+                TransactionAuthFlags::AuthStandard as u8
+            ];
+            auth_standard_bytes.append(&mut spending_condition_bytes.clone());
+
+            let auth_sponsored = TransactionAuth::Sponsored(spending_conditions[i].clone(), spending_conditions[(i+1) % spending_conditions.len()].clone());
+            let mut auth_sponsored_bytes = vec![
+                TransactionAuthFlags::AuthSponsored as u8
+            ];
+            auth_sponsored_bytes.append(&mut spending_condition_bytes.clone());
+            auth_sponsored_bytes.append(&mut spending_condition_2_bytes.clone());
+
+            check_codec_and_corruption::<TransactionAuth>(&auth_standard, &auth_standard_bytes);
+            check_codec_and_corruption::<TransactionAuth>(&auth_sponsored, &auth_sponsored_bytes);
+        }
+    }
+
+    #[test]
+    fn tx_stacks_invalid_spending_conditions() {
```

Comment:
Does this test cover the `VerifyingError("Invalid Message...")`, `VerifyingError("Invalid Signature: failed to decode...")`, and `VerifyingError("Invalid Signature: failed to recover...")` cases? It'd be more clear if this test checked for error equality rather than just `is_err()`.

Reply:
No; I'll add that.